Mathlib/Algebra/Polynomial/Basic.lean

```diff
@@ -482,6 +486,13 @@ theorem monomial_one_right_eq_X_pow (n : ℕ) : monomial n (1 : R) = X ^ n := by
 theorem toFinsupp_X : X.toFinsupp = Finsupp.single 1 (1 : R) :=
   rfl
 
+theorem X_ne_C' (h : (1 : R) ≠ 0) (a : R) : X ≠ C a := by
```

Comment:
Why do you want to have this version? For a `MulZeroOneClass`, `1 ≠ 0` is equivalent to `Nontrivial`.

Reply:
Otherwise LGTM. Please delete this version, then send it to Bors. If you have reasons to preserve this version, then please open a thread on Zulip and tag me there.
bors d+